### ISSUE TITLE
Support for Orchestration service's "stack update" PATCH request verb.

### ISSUE THREAD

**Issue by scott-chaney** [gophercloud/gophercloud#1101]
[Openstack Orchestration API](https://developer.openstack.org/api-ref/orchestration/v1/index.html#stacks)

The Openstack Heat orchestration API distinguishes between PATCH and PUT verbs for the stack update call to afford usage of the existing stack template or enforcement of a specified template being passed in to the call.  This is noted under the "existing" request parameter in the API documentation.

Since the PUT verb requires the template object be specified, it would be useful to have support for the PATCH verb to simply update template's parameters when a template is not available locally.

Note: The Openstack API documentation seems slightly misleading.  The template object is required when using PUT regardless of the --existing request parameter (functionality verified in Mitaka and Queens).

Related Openstack code:
[stack's API update_patch() definition](https://github.com/openstack/heat/blob/master/heat/api/openstack/v1/stacks.py#L494)

also see https://bugs.launchpad.net/python-heatclient/+bug/1224828

**Comment by scott-chaney** [comment]
I am working on this btw.

I wasn't able to find another gophercloud feature that differentiated between the PUT and PATCH for updates.  I implemented the logic in the currnet [Update()](https://github.com/gophercloud/gophercloud/blob/master/openstack/orchestration/v1/stacks/requests.go#L333) function using the "exsiting" flag in the UpdateOpts, similar to the Openstack CLI client pattern.  I thought about breaking it out into an UpdatePatch() function, but that seems more contra to GohperCloud's update pattern.

@jtopjian would you or any of the project maintainers have thoughts on the public method to expose this?

**Comment by jtopjian** [comment]
To make sure I'm understanding this correctly: `UpdatePatch` takes the same options as `Update` but the template and template files are optional?

> I thought about breaking it out into an UpdatePatch() function, but that seems more contra to GopherCloud's update pattern.

If my above assumption is true, I think this is the best implementation. Sure, it creates a second "update" call, but if you omit the name from the design, it's really just any other API call. `UpdatePatch` matches what's in the Heat API code, so that's probably a good name to keep.

The alternative of overloading the existing `Update` function and conditionally calling PUT or PATCH would be the type of design that isn't normal for Gophercloud.

Does that help? Or am I misunderstanding this?

**Comment by scott-chaney** [comment]
Your understanding is correct.  I'll go ahead and create a separate function and get rid of the need for the "existing" flag.

Thanks for the input!

**Comment by scott-chaney** [comment]
@jtopjian I am wondering about the acceptance tests ([stack_tests](https://github.com/gophercloud/gophercloud/blob/master/acceptance/openstack/orchestration/v1/stacks_test.go)) for the orchestration service.  As I started making updates to support the `UpdatePatch` call, I noticed that the existing tests look like they would not work (some invalid refs, structs being built with wrong types, etc.) and from the most [recent successful PR](http://logs.openlabtesting.org/logs/08/1108/1cd649a2c8c7348f651a2472bf71c68280645e37/check/gophercloud-acceptance-test/66eaac0/test_results.html) it appears that the orchestration tests are being skipped, if I am understanding the results correctly.

Any idea of what's going on with those tests?
